OBSDOCS-1414: Release notes for Distributed tracing 3.4 (#83770)
openshift/openshift-docs: observability/distr_tracing/distr-tracing-rn.adoc

include::snippets/distr-tracing-and-otel-disclaimer-about-docs-for-supported-features-only.adoc[]

// git rebase marker 1

[id="distr-tracing_3-4_{context}"]
== Release notes for {DTProductName} 3.4

This release of the {DTProductName} includes the {TempoName} and the deprecated {JaegerName}.

[id="distr-tracing_3-4_cves_{context}"]
=== CVEs

This release fixes the following CVEs:

> * link:https://access.redhat.com/security/cve/CVE-202?-????/[CVE-202?-????]

Reviewer: ```suggestion
* link:https://access.redhat.com/security/cve/CVE-2024-43796[CVE-2024-43796]
* link:https://access.redhat.com/security/cve/CVE-2024-43799[CVE-2024-43799]
* link:https://access.redhat.com/security/cve/CVE-2024-43800[CVE-2024-43800]
* link:https://access.redhat.com/security/cve/CVE-2024-45296[CVE-2024-45296]
* link:https://access.redhat.com/security/cve/CVE-2024-45590[CVE-2024-45590]
* link:https://access.redhat.com/security/cve/CVE-2024-45811[CVE-2024-45811]
* link:https://access.redhat.com/security/cve/CVE-2024-45812[CVE-2024-45812]
* link:https://access.redhat.com/security/cve/CVE-2024-47068[CVE-2024-47068]
* link:https://bugzilla.redhat.com/show_bug.cgi?id=2312579[npm-serialize-javascript: Cross-site Scripting (XSS) in serialize-javascript]
```

I don't know why some links don't work though... :confused: 

For example CVE-2024-45296 is also linked from https://access.redhat.com/errata/RHSA-2024:8014 (errata of another product).

Author: TY. Applied.

[id="distr-tracing_3-4_jaeger-release-notes_{context}"]
=== {JaegerName}

The {JaegerName} 3.4 is provided through the link:https://catalog.redhat.com/software/containers/rhosdt/jaeger-operator-bundle/613b50f3a052bb39f2c5f31e/history[{JaegerOperator} Operator <x>.<y>.<z>].

[NOTE]
====
The {JaegerName} 3.4 is based on the open source link:https://www.jaegertracing.io/[Jaeger] release 1.<y>.0.
====

[IMPORTANT]
====
Jaeger does not use FIPS validated cryptographic modules.
====

[id="distr-tracing_3-4_jaeger-release-notes_support-for-elasticsearch-operator_{context}"]
==== Support for the {es-op}

The {JaegerName} 3.4 is supported for use with the {es-op} 5.6, 5.7, and 5.8.

[id="distr-tracing_3-4_jaeger-release-notes_deprecated-functionality_{context}"]
==== Deprecated functionality

In the {DTProductName} 3.4, Jaeger and support for Elasticsearch remain deprecated, and both are planned to be removed in a future release.
Red Hat will provide support for these components and fixes for CVEs and bugs with critical and higher severity during the current release lifecycle, but these components will no longer receive feature enhancements.

> The {JaegerOperator} Operator will be removed from the `redhat-operators` catalog in November 2025.

Author: @jgomezselles, is this sentence enough or do you still want me to include the link to this page: https://access.redhat.com/solutions/7083722?

Reviewer 2: Thanks! If it is a common practice to link these articles, I think it won't hurt adding the link because it provides more context to users that need more context. If adding the link is an issue, I'm totally fine with the sentence. Thanks again!

Author: Thank you for clarifying, Jose. I've added the link to the solution page just as you requested.

* link:https://access.redhat.com/security/cve/CVE-2024-43800[CVE-2024-43800]
* link:https://access.redhat.com/security/cve/CVE-2024-45296[CVE-2024-45296]
* link:https://access.redhat.com/security/cve/CVE-2024-45590[CVE-2024-45590]
* link:https://access.redhat.com/security/cve/CVE-2024-45811[CVE-2024-45811]
* link:https://access.redhat.com/security/cve/CVE-2024-45812[CVE-2024-45812]
* link:https://access.redhat.com/security/cve/CVE-2024-47068[CVE-2024-47068]
* link:https://bugzilla.redhat.com/show_bug.cgi?id=2312579[Cross-site Scripting (XSS) in `serialize-javascript`]

[id="distr-tracing_3-4_tempo-release-notes_{context}"]
=== {TempoName}

The {TempoName} 3.4 is provided through the link:https://catalog.redhat.com/software/containers/rhosdt/tempo-operator-bundle/642c3e0eacf1b5bdbba7654a/history[{TempoOperator} 0.14.1].

[NOTE]
====
The {TempoName} 3.4 is based on the open source link:https://grafana.com/oss/tempo/[Grafana Tempo] 2.6.1.
====

[id="distr-tracing_3-4_tempo-release-notes_new-features-and-enhancements_{context}"]
==== New features and enhancements

This update introduces the following enhancements:

> * The monitor tab in the Jaeger UI for TempoStack instances uses a new default metrics namespace: `traces.span.metrics`. Before this update, the Jaeger UI used an empty namespace. The new `traces.span.metrics` namespace default is also used by the OpenTelemetry Collector 0.113.0. Users can set the empty value for the metrics namespace by using the following field in the `TempoStack` custom resouce: `spec.template.queryFrontend.monitorTab.redMetricsNamespace: ""`.

Author: Thank you for the suggestion. Applied.

+
[WARNING]
====
This is a breaking change. If you are using both the {TempoName} and {OTELName}, you must upgrade to the {OTELName} 3.4 before upgrading to the {TempoName} 3.4.
====

> * New optional `spec.timeout` field in the `TempoStack` and `TempoMonolithic` custom resource definitions for configuring one timeout value for all components, which is `30s` by default.

Author: Thank you. Applied.

====

[id="distr-tracing_3-4_jaeger-release-notes_{context}"]
=== {JaegerName}

The {JaegerName} 3.4 is provided through the link:https://catalog.redhat.com/software/containers/rhosdt/jaeger-operator-bundle/613b50f3a052bb39f2c5f31e/history[{JaegerOperator} Operator 1.62.0].

[NOTE]
====
The {JaegerName} 3.4 is based on the open source link:https://www.jaegertracing.io/[Jaeger] release 1.62.0.
====
// 1.62.0 is fixed until Jaeger's removal

[IMPORTANT]
====
Jaeger does not use FIPS validated cryptographic modules.
====

[id="distr-tracing_3-4_jaeger-release-notes_support-for-elasticsearch-operator_{context}"]
==== Support for the {es-op}

The {JaegerName} 3.4 is supported for use with the {es-op} 5.6, 5.7, and 5.8.

[id="distr-tracing_3-4_jaeger-release-notes_deprecated-functionality_{context}"]
==== Deprecated functionality

In the {DTProductName} 3.4, Jaeger and support for Elasticsearch remain deprecated, and both are planned to be removed in a future release.
Red{nbsp}Hat will provide support for these components and fixes for CVEs and bugs with critical and higher severity during the current release lifecycle, but these components will no longer receive feature enhancements.

> The {JaegerOperator} Operator link:https://access.redhat.com/solutions/7083722[will be removed] from the `redhat-operators` catalog in November 2025. You must xref:../otel/otel-migrating.adoc#dist-tracing-otel-migrating[migrate] to the xref:distr_tracing_tempo/distr-tracing-tempo-installing.adoc#dist-tracing-tempo-installing[{TempoOperator}] and the xref:../otel/otel-installing.adoc#install-otel[{OTELName}] for distributed tracing collection and storage.

Reviewer 3: ```suggestion
The {JaegerOperator} Operator link:https://access.redhat.com/solutions/7083722[will be removed] from the `redhat-operators` catalog in November 2025. You must xref:../../otel/otel-migrating.adoc#dist-tracing-otel-migrating[migrate] to the xref:../../distr_tracing_tempo/distr-tracing-tempo-installing.adoc#dist-tracing-tempo-installing[{TempoOperator}] and the xref:../../otel/otel-installing.adoc#install-otel[{OTELName}] for distributed tracing collection and storage.
```
Friendly reminder: `xrefs` need to walk back to the project root. So, the number of `../` in an `xref` is determined by how many directories an assembly is nested in. In this case, two directories.

Author: > Friendly reminder: `xrefs` need to walk back to the project root. So, the number of `../` in an `xref` is determined by how many directories an assembly is nested in. In this case, two directories.

@michaelryanpeter and @kalexand-rh, isn't relative path expected according to the repo's [`doc_guidelines.adoc`](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#internal-cross-references)?
![image](https://github.com/user-attachments/assets/f82323ff-0b39-41f8-ba78-a9aaab19cac7)

Author: > but you still have to back up to the root the project and then move forward

:slightly_smiling_face: I still don't understand why, @michaelryanpeter.
Are you aware of any toolchain reason for such a requirement?
And is this requirement written somewhere?

Reviewer 3: Maybe some of the confusion is that I am use the term project root to describe your `openshift-docs` directory. 

Due to the way that our scripts convert xrefs into hard coded links for Pv1 and the structure of our repo, xrefs must back up to the project root and then use the complete path to the assembly your cross referencing. I do not know the specifics of why or how the script works, but I have seem broken builds and malformed links in Pantheon builds due to this issue. Many times.

Asciibinder does not have the same issue. It is much better at figuring out what your cross referencing and is more flexible in this regard.

TLDR: due to the way our repo is structured, you have to back up to your `openshift-docs` folder using the `../` prefix and then add the complete path from your `openshift-docs` folder to the assembly you are xrefing to.

The requirement is written in the OpenShift docs guidelines here: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#internal-cross-references.

If you find the rule confusing or that it needs clarification, please open an issue or PR and we can discuss it at an upcoming OSDOCS and friends meeting. Or feel free to start a thread in #team-ocp-docs.

I hope this helps.

[id="distr-tracing_3-4_jaeger-release-notes_bug-fixes_{context}"]
==== Bug fixes

This update introduces the following bug fix:

> * Before this update, the Jaeger UI could fail with the *502 - Bad Gateway Timeout* error. After this update, users can configure timeout in ingress annotations. (link:https://issues.redhat.com/browse/TRACING-4238[TRACING-4238])

Author: Thank you for the suggestion. Applied.

include::snippets/distr-tracing-and-otel-disclaimer-about-docs-for-supported-features-only.adoc[]

[id="distr-tracing_3-4_{context}"]
== Release notes for {DTProductName} 3.4

This release of the {DTProductName} includes the {TempoName} and the deprecated {JaegerName}.

[id="distr-tracing_3-4_cves_{context}"]
=== CVEs

This release fixes the following CVEs:

* link:https://access.redhat.com/security/cve/CVE-2024-43796[CVE-2024-43796]
* link:https://access.redhat.com/security/cve/CVE-2024-43799[CVE-2024-43799]
* link:https://access.redhat.com/security/cve/CVE-2024-43800[CVE-2024-43800]
* link:https://access.redhat.com/security/cve/CVE-2024-45296[CVE-2024-45296]
* link:https://access.redhat.com/security/cve/CVE-2024-45590[CVE-2024-45590]
* link:https://access.redhat.com/security/cve/CVE-2024-45811[CVE-2024-45811]
* link:https://access.redhat.com/security/cve/CVE-2024-45812[CVE-2024-45812]
* link:https://access.redhat.com/security/cve/CVE-2024-47068[CVE-2024-47068]

> * link:https://bugzilla.redhat.com/show_bug.cgi?id=2312579[Cross-site Scripting (XSS) in `serialize-javascript`]

Reviewer: another resolved CVE, affecting Jaeger UI:
```
* link:https://access.redhat.com/security/cve/CVE-2024-21536[CVE-2024-21536]
```

Author: Thank you. Added.

[id="distr-tracing_3-4_tempo-release-notes_{context}"]
=== {TempoName}

The {TempoName} 3.4 is provided through the link:https://catalog.redhat.com/software/containers/rhosdt/tempo-operator-bundle/642c3e0eacf1b5bdbba7654a/history[{TempoOperator} <x>.<y>.<z>].

[NOTE]

> ====

Author: Thank you. Added.

The {TempoName} 3.4 is based on the open source link:https://grafana.com/oss/tempo/[Grafana Tempo] 2.<y>.0.
====

[id="distr-tracing_3-4_tempo-release-notes_technology-preview-features_{context}"]
==== Technology Preview features

This update introduces the following Technology Preview feature:

* ???.
* ???.
* ???.
* ???.
* ???.
* ???.
* ???.
* ???.
* ???.
* ???.

:FeatureName: Each of these features
include::snippets/technology-preview.adoc[leveloffset=+1]

[id="distr-tracing_3-4_tempo-release-notes_new-features-and-enhancements_{context}"]

> ==== New features and enhancements

Author: @pavolloffay, the linked upstream GitHub release page lists two breaking changes. Should I document both of them here?

> ==== New features and enhancements

Reviewer 4: +1 I forgot about the timeout. The timeout is listed in bugfixes but we can mention it here as well

This update introduces the following enhancements:

* ???.
* ???.
* ???.
* ???.
* ???.
* ???.
* ???.
* ???.
* ???.
* ???.

[id="distr-tracing_3-4_tempo-release-notes_deprecated-functionality_{context}"]
==== Deprecated functionality

* In the {TempoName} 3.4, ???.
* In the {TempoName} 3.4, ???.
* In the {TempoName} 3.4, ???.

[id="distr-tracing_3-4_tempo-release-notes_removal-notice_{context}"]
==== Removal notice

In the {TempoName} 3.4, the FEATURE has been removed. Bug fixes and support are provided only through the end of the 3.? lifecycle. As an alternative to the FEATURE for USE CASE, you can use the ALTERNATIVE instead.

[id="distr-tracing_3-4_tempo-release-notes_bug-fixes_{context}"]
==== Bug fixes

This update introduces the following bug fixes:

> * Before this update, ... . With this update, this issue is fixed by ... . (link:https://issues.redhat.com/browse/TRACING-????[TRACING-????])

Author: @pavolloffay, I don't understand what you mean by
"After this update the query performance for Jaeger UI can be improved by spec.template.queryFrontend.jaegerQuery.findTracesConcurrentRequests field"
Could you please describe it here to me differently?

> * Before this update, ... . With this update, this issue is fixed by ... . (link:https://issues.redhat.com/browse/TRACING-????[TRACING-????])

Author: Responded in https://github.com/openshift/openshift-docs/pull/83770/files#r1837776155.

* Before this update, ... . With this update, this issue is fixed by ... . (link:https://issues.redhat.com/browse/TRACING-????[TRACING-????])
* Before this update, ... . With this update, this issue is fixed by ... . (link:https://issues.redhat.com/browse/TRACING-????[TRACING-????])
* Before this update, ... . With this update, this issue is fixed by ... . (link:https://issues.redhat.com/browse/TRACING-????[TRACING-????])

[id="distr-tracing_3-4_tempo-release-notes_known-issues_{context}"]
==== Known issues

There is currently a known issue:

> * Currently, the {TempoShortName} fails on the {ibm-z-title} (`s390x`) architecture. (link:https://issues.redhat.com/browse/TRACING-3545[TRACING-3545])

Author: Removed.

* Currently, ???. (link:https://issues.redhat.com/browse/TRACING-????[TRACING-????])
* Currently, ???. (link:https://issues.redhat.com/browse/TRACING-????[TRACING-????])
* Currently, ???. (link:https://issues.redhat.com/browse/TRACING-????[TRACING-????])

[id="distr-tracing_3-4_jaeger-release-notes_{context}"]
=== {JaegerName}

> The {JaegerName} 3.4 is provided through the link:https://catalog.redhat.com/software/containers/rhosdt/jaeger-operator-bundle/613b50f3a052bb39f2c5f31e/history[{JaegerOperator} Operator <x>.<y>.<z>].

Author: @pavolloffay, I understand that we will be sticking with one Jaeger version going forward, but could you please confirm which version it is?

Reviewer 4: 3.4 will be based on 1.62.0 cc) @andreasgerstmayr

Reviewer: Yes, we'll stay on Jaeger 1.62.0, which will be in rhosdt-3.4.

Author: Thank you. Fixing the upstream Jaeger version.

However, I still need the Jaeger Operator version.

[NOTE]
====

> The {JaegerName} 3.4 is based on the open source link:https://www.jaegertracing.io/[Jaeger] release 1.<y>.0.

Author: @pavolloffay, I understand that we will be sticking with one Jaeger version going forward, but could you please confirm which version it is?

Author: Copied.

Red{nbsp}Hat will provide support for these components and fixes for CVEs and bugs with critical and higher severity during the current release lifecycle, but these components will no longer receive feature enhancements.

The {JaegerOperator} Operator link:https://access.redhat.com/solutions/7083722[will be removed] from the `redhat-operators` catalog in November 2025. Users must xref:../otel/otel-migrating.adoc#dist-tracing-otel-migrating[migrate] to the xref:distr_tracing_tempo/distr-tracing-tempo-installing.adoc#dist-tracing-tempo-installing[{TempoOperator}] and the xref:../otel/otel-installing.adoc#install-otel[{OTELName}] for distributed tracing collection and storage.

[id="distr-tracing_3-4_jaeger-release-notes_removal-notice_{context}"]
==== Removal notice

In the {JaegerName} 3.4, the FEATURE has been removed. Bug fixes and support are provided only through the end of the 3.? lifecycle. As an alternative to the FEATURE for USE CASE, you can use the ALTERNATIVE instead.

[id="distr-tracing_3-4_jaeger-release-notes_bug-fixes_{context}"]
==== Bug fixes

This update introduces the following bug fixes:

> * Before this update, ???. With this update, ???. (link:https://issues.redhat.com/browse/TRACING-????/[TRACING-????])

Author: Thank you. Added and edited.

* link:https://access.redhat.com/security/cve/CVE-2024-45811[CVE-2024-45811]
* link:https://access.redhat.com/security/cve/CVE-2024-45812[CVE-2024-45812]
* link:https://access.redhat.com/security/cve/CVE-2024-47068[CVE-2024-47068]
* link:https://bugzilla.redhat.com/show_bug.cgi?id=2312579[Cross-site Scripting (XSS) in `serialize-javascript`]

[id="distr-tracing_3-4_tempo-release-notes_{context}"]
=== {TempoName}

The {TempoName} 3.4 is provided through the link:https://catalog.redhat.com/software/containers/rhosdt/tempo-operator-bundle/642c3e0eacf1b5bdbba7654a/history[{TempoOperator} 0.14.1].

[NOTE]
====
The {TempoName} 3.4 is based on the open source link:https://grafana.com/oss/tempo/[Grafana Tempo] 2.6.1.
====

////
[id="distr-tracing_3-4_tempo-release-notes_technology-preview-features_{context}"]
==== Technology Preview features

This update introduces the following Technology Preview feature:

* ???.

:FeatureName: Each of these features
include::snippets/technology-preview.adoc[leveloffset=+1]
////

[id="distr-tracing_3-4_tempo-release-notes_new-features-and-enhancements_{context}"]
==== New features and enhancements

This update introduces the following enhancements:

> * The monitor tab in the Jaeger UI for TempoStack instances uses a new default metrics namespace: `traces.span.metrics`. Before this update, the Jaeger UI used an empty namespace. The new `traces.span.metrics` namespace default is also used by the OpenTelemetry Collector 0.112.0. Users can set the empty value for the metrics namespace by using the following field in the `TempoStack` custom resouce: `spec.template.queryFrontend.monitorTab.redMetricsNamespace: ""`.

Reviewer: ```suggestion
* The monitor tab in the Jaeger UI for TempoStack instances uses a new default metrics namespace: `traces.span.metrics`. Before this update, the Jaeger UI used an empty namespace. The new `traces.span.metrics` namespace default is also used by the OpenTelemetry Collector since version 0.109.0. Users can set the empty value for the metrics namespace by using the following field in the `TempoStack` custom resouce: `spec.template.queryFrontend.monitorTab.redMetricsNamespace: ""`.
```

Author: @andreasgerstmayr, thank you, but I wonder if this (upstream?) info would be confusing to readers because the previous published downstream Collector version we have is `0.107.0` according to https://docs.openshift.com/container-platform/4.14/observability/otel/otel-rn.html#otel_3-3-1_otel-rn.
Perhaps we can leave it as `0.112.0`, which is 3.4, WDYT?

Reviewer: Yes, `0.113.0` is also fine :+1:

+
[WARNING]
====

> This is a breaking change.

Author: @andreasgerstmayr, WDYT? Or feel free to add a suggestion. 
```suggestion
This is a breaking change. Users who are using both the {TempoName} and {OTELName} must upgrade to the {OTELName} 3.4 before upgrading to the {TempoName} 3.4.
```

Reviewer: sgtm

Author: TY, applying.

* Before this update, the {TempoShortName} failed on the {ibm-z-title} (`s390x`) architecture. With this update, the {TempoShortName} is available for the {ibm-z-title} (`s390x`) architecture. (link:https://issues.redhat.com/browse/TRACING-3545[TRACING-3545])
Before this update, the {TempoShortName} failed on clusters with non-private networks. With this update, customers can deploy the {TempoShortName} on clusters with non-private networks. (link:https://issues.redhat.com/browse/TRACING-4507[TRACING-4507])
* Before this update, the Jaeger UI could fail due to reaching a trace quantity limit, resulting in the *504 Gateway Time-out* error in the `tempo-query` logs. After this update, the issue is resolved by introducing two optional fields in the `tempostack` or `tempomonolithic` custom resource:
+
** New `spec.timeout` field for configuring the timeout.
** New `spec.template.queryFrontend.jaegerQuery.findTracesConcurrentRequests` field for improving the query performance of the Jaeger UI.
+
[TIP]
====
One querier can handle up to 20 concurrent queries by default. Increasing the number of concurrent queries further is achieved by scaling up the querier instances.

> ====

Author: @pavolloffay, could you please review this? I rewrote your draft in https://github.com/openshift/openshift-docs/pull/83770/files#r1832705037, but I have little confidence in my wording. Perhaps some things need to be corrected. WDYT?

In the {DTProductName} 3.3, Jaeger and support for Elasticsearch remain deprecated, and both are planned to be removed in a future release.
Red{nbsp}Hat will provide support for these components and fixes for CVEs and bugs with critical and higher severity during the current release lifecycle, but these components will no longer receive feature enhancements.

> The {JaegerOperator} Operator link:https://access.redhat.com/solutions/7083722[will be removed] from the `redhat-operators` catalog in November 2025. Users must xref:../otel/otel-migrating.adoc#dist-tracing-otel-migrating[migrate] to the xref:distr_tracing_tempo/distr-tracing-tempo-installing.adoc#dist-tracing-tempo-installing[{TempoOperator}] and the xref:../otel/otel-installing.adoc#install-otel[{OTELName}] for distributed tracing collection and storage.

Author: Thank you for the suggestion. In this case, Engineering is communicating a specific time frame to customers.

> The {JaegerOperator} Operator link:https://access.redhat.com/solutions/7083722[will be removed] from the `redhat-operators` catalog in November 2025. Users must xref:../otel/otel-migrating.adoc#dist-tracing-otel-migrating[migrate] to the xref:distr_tracing_tempo/distr-tracing-tempo-installing.adoc#dist-tracing-tempo-installing[{TempoOperator}] and the xref:../otel/otel-installing.adoc#install-otel[{OTELName}] for distributed tracing collection and storage.

Reviewer 3: If you want to communicate a specific time, you need to get approval from PX. If engineering misses a deadline, then this could have legal ramifications. My understanding is that this guidance comes from PX and legal.

@kalexand-rh Would you please correct me if I am wrong?

> The {JaegerOperator} Operator link:https://access.redhat.com/solutions/7083722[will be removed] from the `redhat-operators` catalog in November 2025. Users must xref:../otel/otel-migrating.adoc#dist-tracing-otel-migrating[migrate] to the xref:distr_tracing_tempo/distr-tracing-tempo-installing.adoc#dist-tracing-tempo-installing[{TempoOperator}] and the xref:../otel/otel-installing.adoc#install-otel[{OTELName}] for distributed tracing collection and storage.

Reviewer 3: Please see my previous comments regarding how far to back up your xrefs.

> The {JaegerOperator} Operator link:https://access.redhat.com/solutions/7083722[will be removed] from the `redhat-operators` catalog in November 2025. Users must xref:../otel/otel-migrating.adoc#dist-tracing-otel-migrating[migrate] to the xref:distr_tracing_tempo/distr-tracing-tempo-installing.adoc#dist-tracing-tempo-installing[{TempoOperator}] and the xref:../otel/otel-installing.adoc#install-otel[{OTELName}] for distributed tracing collection and storage.

Reviewer 5: +1 - we shouldn't merge this without additional approval/guarantees or removing the commitment to a specific date.

> The {JaegerOperator} Operator link:https://access.redhat.com/solutions/7083722[will be removed] from the `redhat-operators` catalog in November 2025. Users must xref:../otel/otel-migrating.adoc#dist-tracing-otel-migrating[migrate] to the xref:distr_tracing_tempo/distr-tracing-tempo-installing.adoc#dist-tracing-tempo-installing[{TempoOperator}] and the xref:../otel/otel-installing.adoc#install-otel[{OTELName}] for distributed tracing collection and storage.

Author: @michaelryanpeter and @kalexand-rh, I have added the word `plan` to comply with the quoted https://redhat-documentation.github.io/supplementary-style-guide/#statements-about-the-future:
```
The {JaegerOperator} Operator link:https://access.redhat.com/solutions/7083722
[is planned to be removed] from the `redhat-operators` catalog in November 2025.
```
![image](https://github.com/user-attachments/assets/8d5536df-8887-43d7-bc5c-a6d2e2897829)

Regarding:
> +1 - we shouldn't merge this without additional approval/guarantees or removing the commitment to a specific date.

@pavolloffay is making the commitment on behalf of the engineering team in his next comment in this thread.

> The {JaegerOperator} Operator link:https://access.redhat.com/solutions/7083722[will be removed] from the `redhat-operators` catalog in November 2025. Users must xref:../otel/otel-migrating.adoc#dist-tracing-otel-migrating[migrate] to the xref:distr_tracing_tempo/distr-tracing-tempo-installing.adoc#dist-tracing-tempo-installing[{TempoOperator}] and the xref:../otel/otel-installing.adoc#install-otel[{OTELName}] for distributed tracing collection and storage.

Author: > +1 - we shouldn't merge this without additional approval/guarantees or removing the commitment to a specific date.

@kalexand-rh, who do we need "additional approval/guarantees" from? The engineering team stakeholders have already approved this.

> The {JaegerOperator} Operator link:https://access.redhat.com/solutions/7083722[will be removed] from the `redhat-operators` catalog in November 2025. Users must xref:../otel/otel-migrating.adoc#dist-tracing-otel-migrating[migrate] to the xref:distr_tracing_tempo/distr-tracing-tempo-installing.adoc#dist-tracing-tempo-installing[{TempoOperator}] and the xref:../otel/otel-installing.adoc#install-otel[{OTELName}] for distributed tracing collection and storage.

Author: @pavolloffay, I have replaced Jose's requested `November 2025` with `in a future release` for the 3.4 release notes. This give us time until 3.5 to clarify and obtain any approvals.
